src/Microsoft.AspNetCore.SignalR.Client.Core/HubConnection.cs

```diff
@@ -132,24 +132,25 @@ public IDisposable On(string methodName, Type[] parameterTypes, Func<object[], o
                 // Start the connection
                 var connection = _connectionFactory();
                 await connection.StartAsync(_protocol.TransferFormat);
-                _connectionState = new ConnectionState(connection, this);
+                var connectionState = new ConnectionState(connection, this);
```

Comment:
Maybe call this `startingConnectionState` to make it more obvious that it isn't ready to use yet?